### PR TITLE
Fix nil pointer error on dmsghttp config with offline stun server

### LINKED ISSUE #1274: nil pointer startup visor on dmsghttp config
**Describe the bug**
Currently if STUN servers be empty, or all of them not available, then we get panic on visor startup due these logs:
```
[2022-06-30T03:18:23+04:30] WARN [stun_client]: Reply from 172.104.188.139:3478 server: UDP is blocked
[2022-06-30T03:18:23+04:30] WARN [stun_client]: All STUN servers offline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xa07b21]

goroutine 28 [running]:
github.com/ccding/go-stun/stun.(*Host).IP(...)
	/home/mohammed/Projects/Skycoin/skywire/vendor/github.com/ccding/go-stun/stun/host.go:54
github.com/skycoin/skywire/pkg/visor.getPublicIP(0xc0001345a0, {0xc00023caa0, 0x4c})
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:1185 +0xe1
github.com/skycoin/skywire/pkg/visor.initAddressResolver({0x10be6a0, 0xc0001b0690}, 0xc0001345a0, 0xc0000381e0)
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:213 +0x97
github.com/skycoin/skywire/pkg/visor.withInitCtx.func1({0x10be6a0, 0xc0001b0690}, 0xc0001b0690)
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:1119 +0xb9
github.com/skycoin/skywire/pkg/visor/visorinit.(*Module).InitConcurrent(0x165c180, {0x10be6a0, 0xc0001b0690})
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/visorinit/module.go:136 +0x278
created by github.com/skycoin/skywire/pkg/visor/visorinit.(*Module).InitConcurrent
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/visorinit/module.go:115 +0x11c
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xa07b21]

goroutine 29 [running]:
github.com/ccding/go-stun/stun.(*Host).IP(...)
	/home/mohammed/Projects/Skycoin/skywire/vendor/github.com/ccding/go-stun/stun/host.go:54
github.com/skycoin/skywire/pkg/visor.getPublicIP(0xc0001345a0, {0xc00023cc30, 0x4c})
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:1185 +0xe1
github.com/skycoin/skywire/pkg/visor.initDiscovery({0x10be6a0, 0xc0001b0690}, 0xc0001345a0, 0xc0000381e0)
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:257 +0x1d5
github.com/skycoin/skywire/pkg/visor.withInitCtx.func1({0x10be6a0, 0xc0001b0690}, 0xc0001b0690)
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/init.go:1119 +0xb9
github.com/skycoin/skywire/pkg/visor/visorinit.(*Module).InitConcurrent(0x165c240, {0x10be6a0, 0xc0001b0690})
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/visorinit/module.go:136 +0x278
created by github.com/skycoin/skywire/pkg/visor/visorinit.(*Module).InitConcurrent
	/home/mohammed/Projects/Skycoin/skywire/pkg/visor/visorinit/module.go:115 +0x11c

```

### PR DESCRIPTION
**Pull request by mrpalide**
Did you run `make format && make check`? Yes

Fixes #1274 #1044

 Changes:	
- return error when all stun servers was offline

How to test this PR:
- `make build`
- generate dsmghttp config `skywire-cli config gen -dit`
- put arbitrary URL on stun_servers value in config, like:
  ```
  "stun_servers": [
	"172.112.20.3:3478"
  ],
  ```
- start visor and check logs

### CODE REVIEW

File: pkg/visor/init.go [hunk 1]
```diff
@@ -1179,9 +1182,35 @@ func getPublicIP(v *Visor, service string) (pIP string, err error) {
...
-	pIP = v.stunClient.PublicIP.IP()
 	return pIP, nil
 }
+
+type ipAPI struct {
+	Query string
+}
+
+func getip2() (string, error) {
+	req, err := http.Get("http://ip-api.com/json/")
```
Comment (ersonp) [inline pkg/visor/init.go]: we should use our own service here `ip.skycoin.com`

File: pkg/visor/init.go [hunk 2]
```diff
@@ -1179,9 +1182,35 @@ func getPublicIP(v *Visor, service string) (pIP string, err error) {
...
 
-	pIP = v.stunClient.PublicIP.IP()
 	return pIP, nil
 }
+
+type ipAPI struct {
+	Query string
+}
+
+func getip2() (string, error) {
```
Comment (ersonp) [inline pkg/visor/init.go]: The func name can be `getIP()`

**Comment by ersonp** [comment]
I think we should take the IP from stun client as a fallback for when the IP API fails. If the stun servers are working we get the IP anyway.

File: pkg/visor/init.go (hunk 2 above)
Comment (mrpalide) [reply pkg/visor/init.go]: Fixed.

File: pkg/visor/init.go (hunk 1 above)
Comment (mrpalide) [reply pkg/visor/init.go]: Fixed.

File: pkg/visor/init.go [hunk 3]
```diff
@@ -1179,9 +1182,43 @@ func getPublicIP(v *Visor, service string) (pIP string, err error) {
 		return pIP, fmt.Errorf("provided URL is invalid: %w", err)
 	}
 
-	// Wait until stun client is ready
-	<-v.stunReady
+	pIP, err = getIP()
```
Comment (ersonp) [inline pkg/visor/init.go:1185]: This returns empty sting.

Comment (mrpalide) [reply pkg/visor/init.go:1185]: Fixed.

**Review by ersonp — APPROVED** [review]

**Review by jdknives — APPROVED** [review]

**Pull request merged**